pkg/disk/sfdisk/expand_test.go

```diff
@@ -6,36 +6,70 @@ import (
 	"errors"
 	"os"
 	"os/exec"
+	"strconv"
 	"strings"
 	"testing"
 
 	utilsos "github.com/kubernetes-sigs/alibaba-cloud-csi-driver/pkg/utils/os"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
+func runCommand(t *testing.T, name string, arg ...string) string {
+	cmd := exec.Command(name, arg...)
+	output, err := cmd.Output()
+	require.NoError(t, utilsos.ErrWithStderr(err))
+	return string(output)
+}
+
 func TestExpandPartition(t *testing.T) {
 	path, err := exec.LookPath("sfdisk")
 	if errors.Is(err, exec.ErrNotFound) {
 		t.Skip("sfdisk not found")
 	}
-	assert.NoError(t, err)
+	require.NoError(t, err)
 	t.Logf("sfdisk found at: %s", path)
 
 	testImage := t.TempDir() + "/test.img"
-	_, err = os.Create(testImage)
-	assert.NoError(t, err)
-	assert.NoError(t, os.Truncate(testImage, 1<<23)) // 8MB
+	f, err := os.Create(testImage)
+	require.NoError(t, err)
+	require.NoError(t, f.Close())
+	require.NoError(t, os.Truncate(testImage, 1<<23)) // 8MB
 
 	cmd := exec.Command("sfdisk", testImage)
 	cmd.Stdin = bytes.NewReader([]byte("label: gpt\n,\n")) // create a single partition
 	result, err := cmd.Output()
-	assert.NoError(t, utilsos.ErrWithStderr(err))
+	require.NoError(t, utilsos.ErrWithStderr(err))
 	t.Logf("create partition success: %s", string(result))
 
-	assert.NoError(t, os.Truncate(testImage, 1<<24)) // expand to 16MB
-	assert.NoError(t, ExpandPartition(context.Background(), testImage, "1"))
+	result, err = exec.Command("losetup", "--find", testImage, "--show", "--partscan").Output()
```

Comment:
Can we add a concurrency test to verify the effectiveness of the lock parameters?

Reply:
`--lock` is actually intended to protect us from concurrent access with udev.